src/lualib/string.d.ts

```diff
@@ -1,9 +1,11 @@
 /** @luaIterator */
-interface GMatchResult extends Iterable<string> { }
+interface GMatchResult extends Array<string> { }
 
+/** @noSelf */
 declare namespace string {
     /** @tupleReturn */
     function gsub(source: string, searchValue: string, replaceValue: string): [string, number];
+    function gsub(source: string, searchValue: string, replaceValue: (...groups: string[]) => string): [string, number];
```

Comment:
this will break when my tupleReturn PR is merged - you'll need to specify `@tupleReturn` for each overload

Reply:
Fixed this, diff doesn't seem to pick it up